replica/table.cc

```diff
@@ -1496,6 +1496,7 @@ bool table::can_flush() const {
 }
 
 future<> table::clear() {
+    auto permits = co_await _config.dirty_memory_manager->get_all_flush_permits();
     if (_commitlog) {
```

Comment:
Strictly speaking maybe this should be the first patch.

Reply:
I can do that, though it makes reproducing https://github.com/scylladb/scylla/issues/10421 harder.